sysdig/internal/client/monitor/models_v2.go

```diff
@@ -0,0 +1,173 @@
+package monitor
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+)
+
+// --- Constants --
+const (
+	// alert types enum
+	AlertV2AlertType_AdvancedManual   = "ADVANCED_MANUAL"
+	AlertV2AlertType_AnomalyDetection = "ANOMALY_DETECTION"
+	AlertV2AlertType_Dowtime          = "DOWNTIME"
+	AlertV2AlertType_Event            = "EVENT"
+	AlertV2AlertType_GroupOutlier     = "GROUP_OUTLIER"
+	AlertV2AlertType_Manual           = "MANUAL"
+	AlertV2AlertType_Prometheus       = "PROMETHEUS"
+
+	// severities enum
+	AlertV2Severity_High   = "high"
+	AlertV2Severity_Medium = "medium"
+	AlertV2Severity_Low    = "low"
+	AlertV2Severity_Info   = "info"
+
+	// alert link type
+	AlertLinkV2Type_Dashboard = "dashboard"
+	AlertLinkV2Type_Runbook   = "runbook"
+
+	// others
+	AlertV2CaptureFilenameRegexp = `.*?\.scap`
+)
+
+// enums severity values
+func AlertV2Severity_Values() []string {
+	return []string{
+		AlertV2Severity_High,
+		AlertV2Severity_Medium,
+		AlertV2Severity_Low,
+		AlertV2Severity_Info,
+	}
+}
+
+// AlertV2
+type AlertV2Common struct {
+	ID                            int                            `json:"id,omitempty"`
+	Version                       int                            `json:"version,omitempty"`
+	Name                          string                         `json:"name"`
+	Description                   string                         `json:"description,omitempty"`
+	DurationSec                   int                            `json:"durationSec"`
+	Type                          string                         `json:"type"`
+	Group                         string                         `json:"group,omitempty"`
+	Severity                      string                         `json:"severity"`
+	TeamID                        int                            `json:"teamId,omitempty"`
+	Enabled                       bool                           `json:"enabled"`
+	NotificationChannelConfigList *[]NotificationChannelConfigV2 `json:"notificationChannelConfigList,omitempty"`
+	CustomNotificationTemplate    *CustomNotificationTemplateV2  `json:"customNotificationTemplate,omitempty"`
+	CaptureConfig                 *CaptureConfigV2               `json:"captureConfig,omitempty"`
+	Links                         *[]AlertLinkV2                 `json:"links,omitempty"`
```

Comment:
@arturodilecce I dont' see any check on Links nil pointer, is this something we can avoid?

Reply:
@mbovo atm the AlertV2<type> resources does not support configuring Links as the API are not well documented about it (I guess it's still an experimental feature). I've added the attribute into the AlertV2Common type for future use.